cdap-watchdog/src/main/java/co/cask/cdap/logging/save/LogSaver.java

```diff
@@ -135,17 +141,19 @@ private void createProcessors(Set<Integer> partitions) throws Exception {
     }
   }
 
-  private void cancelLogCollectorCallbacks() {
+  private Set<Cancellable> cancelLogCollectorCallbacks() {
+    Set<Cancellable> cancellables = new HashSet<>();
    for (Entry<Integer, Cancellable> entry : kafkaCancelMap.entrySet()) {
       if (entry.getValue() != null) {
         LOG.info("Cancelling kafka callback for partition {}", entry.getKey());
         kafkaCancelCallbackLatchMap.get(entry.getKey()).countDown();
-        entry.getValue().cancel();
+        cancellables.add(entry.getValue());
       }
     }
 
     kafkaCancelMap.clear();
     kafkaCancelCallbackLatchMap.clear();
```

Comment:
Also we can move this clearing logic to place after which we actually cancel the callbacks.